### PR TITLE
Snapshot config to define column names

### LINKED ISSUE #6112: [Core] New `snapshot_meta_column_names` config for dbt snapshots
At a minimum, update body to include a link to the page on docs.getdbt.com requiring updates and what part(s) of the page you would like to see updated.
Originating from this issue: https://github.com/dbt-labs/dbt-core/issues/10185
Pull request: https://github.com/dbt-labs/dbt-core/pull/10608

### PR DESCRIPTION
**Pull request by nataliefiann**
I've created this PR / added section to highlight new config allows users to define column names following this Git issue: https://github.com/orgs/dbt-labs/projects/14/views/35?pane=issue&itemId=80581863

Resolves [#6112](https://github.com/dbt-labs/docs.getdbt.com/issues/6112)

## Checklist
- [x] I have reviewed the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) so my content adheres to these guidelines.
- [x] The topic I'm writing about is for specific dbt version(s) and I have versioned it according to the [version a whole page](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) and/or [version a block of content](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-blocks-of-content) guidelines.
- [ ] I have added checklist item(s) to this list for anything anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."

### CODE REVIEW

File: website/docs/reference/snapshot-configs.md [hunk 1]
```diff
@@ -301,3 +301,48 @@ snapshots:
 ```
 
 </File>
+
+<VersionBlock firstVersion="1.9">
+
+#### Define column names in snapshot table
+
+You can use the following config to define the names of the columns that are created in the [snapshot table](/docs/build/snapshots#snapshot-meta-fields).
```
Comment (mirnawong1) [inline website/docs/reference/snapshot-configs.md]:
```suggestion
Use the following configuration to define the names of the columns created in the [snapshot table](/docs/build/snapshots#snapshot-meta-fields).
```

File: website/docs/reference/snapshot-configs.md [hunk 2]
```diff
@@ -301,3 +301,48 @@ snapshots:
...
+      dbt_valid_from: example_valid_from,
+      dbt_valid_to: example_valid_to
+    }
+
+```
+
+</File>
+
+- When you run your snapshots, dbt handles the renames from `dbt_valid_from` to `example_valid_from` for you.
+- Update your config to:
```
Comment (mirnawong1) [inline website/docs/reference/snapshot-configs.md]:
```suggestion
- If you make subsequent updates to the config to rename the column, dbt manages the migration of column names for each configuration change. For example:
```

File: website/docs/reference/snapshot-configs.md [hunk 3]
```diff
@@ -301,3 +301,48 @@ snapshots:
...
+      dbt_valid_to: test_valid_to
+    }
+
+```
+
+</File>
+
+- When you run your snapshots, dbt handles the renames from `example_valid_from` to `test_valid_from` for you.
+
+You can read through the section on [Defining column names](/docs/build/snapshots#define-column-names-in-snapshot-table) for more information.
```
Comment (mirnawong1) [inline website/docs/reference/snapshot-configs.md:346]: the link doesn't take me to the right place? or for some reason i didn't see the 'define column names' section?

File: website/docs/reference/snapshot-configs.md [hunk 4]
```diff
@@ -301,3 +301,48 @@ snapshots:
 ```
 
 </File>
+
+<VersionBlock firstVersion="1.9">
+
+#### Define column names in snapshot table
```
Comment (graciegoheen) [inline website/docs/reference/snapshot-configs.md]:
```suggestion
#### Customize meta column names in snapshot table
```

File: website/docs/reference/snapshot-configs.md [hunk 5]
```diff
@@ -301,3 +301,48 @@ snapshots:
...
+
+Use the following configuration to define the names of the columns created in the [snapshot table](/docs/build/snapshots#snapshot-meta-fields).
```
Comment (graciegoheen) [inline website/docs/reference/snapshot-configs.md]:
```suggestion
Use the following configuration to customize the names of the meta columns dbt creates in the [snapshot table](/docs/build/snapshots#snapshot-meta-fields).
```

File: website/docs/reference/snapshot-configs.md [hunk 6]
```diff
@@ -301,3 +301,48 @@ snapshots:
...
+
+<File name='dbt_project.yml'>
+
+```yml
+
+snapshots:
```
Comment (graciegoheen) [inline website/docs/reference/snapshot-configs.md:315]: Let's use the full example from the issue!

```
snapshots:
  my_project:
    +snapshot_meta_column_names={
      dbt_valid_from: start_date,
      dbt_valid_to: end_date,
      dbt_scd_id: scd_id,
      dbt_updated_at: updated_date
    }
```

File: website/docs/reference/snapshot-configs.md [hunk 7]
```diff
@@ -301,3 +301,48 @@ snapshots:
...
 
 </File>
+
+<VersionBlock firstVersion="1.9">
+
+#### Define column names in snapshot table
+
+Use the following configuration to define the names of the columns created in the [snapshot table](/docs/build/snapshots#snapshot-meta-fields).
+
+<File name='dbt_project.yml'>
```
Comment (graciegoheen) [inline website/docs/reference/snapshot-configs.md:311]: let's also show the example from the issue on how to do this in the config block:

```
{{
    config(
      target_database='analytics',
      target_schema='snapshots',
      unique_key='id',

      strategy='timestamp',
      updated_at='updated_at',
      
      snapshot_meta_column_names={
        dbt_valid_from: start_date,
        dbt_valid_to: end_date,
        dbt_scd_id: scd_id,
        dbt_updated_at: updated_date
      }
    )
}}
```

Comment (graciegoheen) [reply website/docs/reference/snapshot-configs.md:311]: we have some tab thing for other examples so you can see how to write a config in different places

File: website/docs/reference/snapshot-configs.md [hunk 8]
```diff
@@ -301,3 +301,48 @@ snapshots:
...
+      dbt_valid_from: example_valid_from,
+      dbt_valid_to: example_valid_to
+    }
+
+```
+
+</File>
+
+- When you run your snapshots, dbt handles the renames from `dbt_valid_from` to `example_valid_from` for you.
+- If you make subsequent updates to the config to rename the column, dbt manages the migration of column names for each configuration change. For example:
```
Comment (graciegoheen) [inline website/docs/reference/snapshot-configs.md:327]: This is not true.

See issue -> https://github.com/dbt-labs/docs.getdbt.com/issues/6112#issuecomment-2374659927

> we aren't doing anything for migration, so if a user renames the columns for a snapshot without updating an existing table, they will get an error. The recommendation is to use these settings for new snapshots, or to arrange to update their existing tables prior to committing a column name change.

File: website/docs/reference/snapshot-configs.md [hunk 9]
```diff
@@ -301,3 +301,48 @@ snapshots:
...
+    +snapshot_meta_column_names ={
+      dbt_valid_from: test_valid_from,
+      dbt_valid_to: test_valid_to
+    }
+
+```
+
+</File>
+
+- When you run your snapshots, dbt handles the renames from `example_valid_from` to `test_valid_from` for you.
```
Comment (graciegoheen) [inline website/docs/reference/snapshot-configs.md:344]: this is not true

Comment (graciegoheen) [inline website/docs/reference/snapshot-configs.md:344]: same comment as above - this is not true

**Review by graciegoheen — CHANGES_REQUESTED** [review]
Left some comments - I think this needs more work. Please see docs issue for examples to include and callout that we do **not** handle the migration -> https://github.com/dbt-labs/docs.getdbt.com/issues/6112#issuecomment-2374659927

**Comment by nataliefiann** [comment]
Closed off as Core project has been paused

**Pull request closed without merging**